ui/packages/ui/src/app/components/formHelpers/MaskOrTruncateItem.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+  Flex,
+  FlexItem,
+  Grid,
+  GridItem,
+  TextInput,
+} from "@patternfly/react-core";
+import { MinusCircleIcon } from "@patternfly/react-icons";
+import * as React from "react";
+import "./MaskOrTruncateItem.css";
+
+export interface IMaskOrTruncateItemProps {
+  rowId: number;
+  columnsValue?: string;
+  nValue?: string;
+  canDelete: boolean;
+  maskTruncateItemChanged: (rowId: number, maskTruncateValue: string) => void;
+  deleteMaskTruncateItem: (rowId: number) => void;
+}
+
+export const MaskOrTruncateItem: React.FunctionComponent<IMaskOrTruncateItemProps> = (
+  props
+) => {
+
+  const [currentColumns, setCurrentColumns] = React.useState<string | undefined>(props.columnsValue);
```

Comment:
for the React components initialized state from `props`, we should add the corresponding `useEffect` code to update the state value in case the props passed value is being changed.